### PR TITLE
Ibmid saas flag

### PR DESCRIPTION
**Pull request by rashmi43**

### CODE REVIEW

File: pkg/apis/operator/v1alpha1/authentication_types.go [hunk 1]
```diff
@@ -104,6 +104,7 @@ type ConfigSpec struct {
 	ICPPort                     int32  `json:"icpPort"`
 	FIPSEnabled                 bool   `json:"fipsEnabled"`
 	ROKSEnabled                 bool   `json:"roksEnabled"`
+	IBMCloudSass                bool   `json:"ibmCloudSass"`
 	NONCEEnabled                bool   `json:"nonceEnabled"`
```
Comment (posriniv) [inline pkg/apis/operator/v1alpha1/authentication_types.go:108]: please add omitempty here @rashmi43

**Review by posriniv — APPROVED** [review]
lgtm

File: pkg/apis/operator/v1alpha1/authentication_types.go [hunk 2]
```diff
@@ -104,6 +104,7 @@ type ConfigSpec struct {
 	ICPPort                     int32  `json:"icpPort"`
 	FIPSEnabled                 bool   `json:"fipsEnabled"`
 	ROKSEnabled                 bool   `json:"roksEnabled"`
+	IBMCloudSaas                bool   `json:"ibmCloudSaas,omitempty"`
 	NONCEEnabled                bool   `json:"nonceEnabled"`
```
Comment (posriniv) [inline pkg/apis/operator/v1alpha1/authentication_types.go:109]: @rashmi43 please add your variable here too

File: pkg/controller/authentication/configmap.go [hunk 3]
```diff
@@ -271,6 +278,7 @@ func (r *ReconcileAuthentication) authIdpConfigMap(instance *operatorv1alpha1.Au
 			"FIPS_ENABLED":                       strconv.FormatBool(instance.Spec.Config.FIPSEnabled),
 			"NONCE_ENABLED":                      strconv.FormatBool(instance.Spec.Config.NONCEEnabled),
 			"ROKS_ENABLED":                       strconv.FormatBool(instance.Spec.Config.ROKSEnabled),
+			"IBM_CLOUD_SAAS":                     strconv.FormatBool(instance.Spec.Config.IBMCloudSaas),
 			"ROKS_URL":                           instance.Spec.Config.ROKSURL,
```
Comment (posriniv) [inline pkg/controller/authentication/configmap.go:283]: here too

**Review by posriniv — APPROVED** [review]
lgtm

**Review by yannizhang2019 — APPROVED** [review]

**Review by pnpavan25 — APPROVED** [review]
lgtm

**Pull request merged**